src/lib.rs

```diff
@@ -106,16 +110,34 @@ impl Optimizer {
     pub fn parse_bril(program: &str) -> Result<Program, EggCCError> {
         let abstract_prog =
             parse_abstract_program_from_read(program.as_bytes(), false, false, None);
-        let serialized = serde_json::to_string(&abstract_prog).unwrap();
 
-        // call SSA conversion
+        // TODO dumb encoding does not support phi nodes yet
+        /*
+        let serialized = serde_json::to_string(&abstract_prog).unwrap();
         let ssa_output = run_command_with_stdin(
             std::process::Command::new("python3").arg("bril/examples/to_ssa.py"),
             serialized,
         );
         let ssa_prog: AbstractProgram = serde_json::from_str(&ssa_output).unwrap();
 
-        Program::try_from(ssa_prog).map_err(|err| EggCCError::ConversionError(err.to_string()))
+        let ssa_res = Program::try_from(ssa_prog)
+            .map_err(|err| EggCCError::ConversionError(err.to_string()))?;
+
+        let dead_code_optimized = run_command_with_stdin(
+            std::process::Command::new("python3").arg("bril/examples/tdce.py"),
+            serde_json::to_string(&ssa_res).unwrap(),
+        );
+
+        let res: Program = serde_json::from_str(&dead_code_optimized)
+            .map_err(|err| EggCCError::ConversionError(err.to_string()))?;
+        */
+
+        let prog = Program::try_from(abstract_prog)
+            .map_err(|err| EggCCError::ConversionError(err.to_string()))?;
+
+        Optimizer::check_for_uninitialized_vars(&prog)?;
```

Comment:
Throw an error for uninitialized variables.
Really just checks for  `__undefined` in the ssa converted program. For now, does nothing.